### PR TITLE
Add missing translations in templates

### PR DESCRIPTION
**Pull request by blgo**
Some titles and buttons where not translated: 
Add pagination translations
Add navBar dynamic titles translations:
  * taxonomies need a i18n translation for display
  * terms do not need a translation because this will be should be
    already translated in the markdown files:
Add translations to category taxonimy templates

### CODE REVIEW

**Comment by statnmap** [comment]
Hi @blgo, thank you for your contribution. I do not understand what `{{ i18n .Section }}` is supposed to do ?

**Comment by blgo** [comment]
I have fixed the bad conditional there. It was a sloppy copy/paste.

Honestly, I just amended that piece of code to support translations and I don't remember what 
"Section" means in this template.

**Comment by statnmap** [comment]
I went through your suggestions. 
Thank you for that.
I agree that we forgot the previous/next page translations.

However, considering the `{{ i18n .Section }}` and `{{ i18n .Title }}` you added, I think the problem is little deeper than that. Indeed, `.Section` name translation normally already exists in the `toml` file. It is the `name = ...`. Thus, we could probably find a way to use this translation without the need to add a new one in the `i18n` file. I tried a few things but it does not work so far.

Your proposition is a good temporary solution, however, if `.Section` names of the default language contains some space, it does not work. We also have to say to add every menu name in the i18n files.
Indeed, `i18n` files should only contain simple strings.

Do you have a proposition that would use the .Section Name instead of the i18n ?

**Comment by blgo** [comment]
I found what .**Section** means and it **does need to be translated**

i.e. https://themes.gohugo.io/theme/future-imperfect/categories/hugo the **.Section** is **Categories**
[https://gohugo.io/content-management/sections/#section-page-variables-and-methods](https://gohugo.io/content-management/sections/#section-page-variables-and-methods)

**.Title,** however, comes from the page's title, these **cannot be translated in most cases.** 

**Because .Section is always available, Title is never used in the Theme.** 

The only scenario where we wouln't translate .Section, would be if we are using "slugs" for translations to have a translated URL as well. 

_Also, the links go to the main page always and possibly the should reference the current page but this is a different issue I will report in the main repo._

This is a master brach issue, however, .Section does need to check for translations.

I have updated the pull request branch.

**Comment by blgo** [comment]
Should I make [this change](https://github.com/blgo/hugo-future-imperfect/blob/ef02a37de799fbb086d173cdc066fdc54195db78/layouts/taxonomy/category.terms.html ) conditional?

**Comment by statnmap** [comment]
My problem is that, for me, .Section returns something different depending on which page you are, otherwise this would not be a variable. It may mean "blog", "categories", "tag", ...
Thus we can not define a fixed translation for that. This may work on your website, but not if other persons decided to add a new Taxonomy or so.

File: layouts/taxonomy/category.terms.html [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
         <ul class="posts">
             <header>
-                <h1>{{ .Data.Plural }}</h1>
+                <h1>{{ i18n .Data.Plural }}</h1>
```
Comment (blgo) [inline layouts/taxonomy/category.terms.html]: NOTE: This is my website and **categories** was replaced by **destinations**, however I didn't change the behavior.

**Before:**
image

**After adding this change:**
image

Should I add fall-back for this as well?

**Comment by blgo** [comment]
I am sorry but I don't undersand _"Thus we can not define a fixed translation for that. This may work on your website, but not if other persons decided to add a new Taxonomy or so."_
I am actually using a different taxonomy in my website... 

Also, **I have found more missing translations on general-tittle.html**

The are several problems here, we have to translate all using config.toml or all using i18n.

I have the feeling that you would like to translate more using the config.toml file. I would much rather using some Go function that would check if there is a translation on **i18n** and if there isn't one, then fallback to the default language.

However, this theme is quite complex (or should I say complicated?) and the Titles, Sections, Taxonomies; names or strings in general, do not propagate through the website in an obvious way.

At this point, this is becoming really complex and I am getting different results during my tests on [the multi-language demo website](https://demo-hugo-future-imperfect-multilanguage.netlify.com/categories/). I am not sure if I am using the wrong Hugo version or this theme is full of complicated **bugs**.

I hope @pescador sees this conversation and shades some light.

**Comment by statnmap** [comment]
Well, I am probably trying to do things more complicated than they are.
I added tags in the translation because there are now in the main repository too.
Concerning the translation of .Section, we can let your `if` statement. For now, I do not know how it work if the name has multiple words separated with space and how we can integrate that in the i18n file.
We need to add in the documentation that if you add a new taxonomy, you need to provide its english name and translations in the i18n files.

**Pull request merged**